oauthenticator/generic.py

```diff
@@ -99,52 +95,71 @@ def user_info_to_username(self, user_info):
 
         return username
 
-    async def user_is_authorized(self, auth_model):
-        user_info = auth_model["auth_state"][self.user_auth_state_key]
-        if self.allowed_groups:
-            self.log.info(
-                f"Validating if user claim groups match any of {self.allowed_groups}"
+    def get_user_groups(self, user_info):
+        """
+        Returns a set of groups the user belongs to based on claim_groups_key
+        and provided user_info.
+
+        - If claim_groups_key is a callable, it is meant to return the groups
+          directly.
+        - If claim_groups_key is a nested dictionary key like
+          "permissions.groups", this function returns
+          user_info["permissions"]["groups"].
+        """
```

Comment:
Hmm, I think we should keep it like this, without a prefix, but I don't have strong preferences.

> Do we see this as relevant for others to make use of?

Maybe it can be used by someone who wants to build their own oauthenticator starting from this generic implementation, and want to add more complex logic related to groups? then I believe it should be clear that it's ok to make use of this?

But again, since it doesn't actually put any constraints on its usage, if you have strong opinions about `_` prefix or not, will go with that.

Thanks @consideRatio! ✨

Reply:
Okay let's go without a prefix! I suggested a comment in https://github.com/jupyterhub/oauthenticator/pull/594/files#r1227872117 to help for example myself figure out this is a public method but not part of the OAuthenticator base class methods overridden.